src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -403,6 +405,124 @@ public async Task CreateAsync(ContentItem contentItem, VersionOptions options)
             }
         }
 
+        public async Task<ContentValidateResult> ValidateAsync(ContentItem contentItem)
+        {
+            var context = new ValidateContentContext(contentItem);
+
+            await Handlers.InvokeAsync((handler, context) => handler.ValidateAsync(context), context, _logger);
+
+            if (!context.ContentValidateResult.Succeeded)
+            {
+                _session.Cancel();
+            }
+
+            return context.ContentValidateResult;
+        }
+
+        public Task<ContentValidateResult> CreateContentItemVersionAsync(ContentItem contentItem)
+        {
+            return CreateContentItemVersionAsync(contentItem, null);
+        }
+
+        public Task<ContentValidateResult> UpdateContentItemVersionAsync(ContentItem updatingVersion, ContentItem updatedVersion)
+        {
+            return UpdateContentItemVersionAsync(updatedVersion, updatedVersion);
+        }
+
+        public async Task ImportAsync(IEnumerable<ContentItem> contentItems)
+        {
+            var skip = 0;
+            var take = _importBatchSize;
+
+            var batchedContentItems = contentItems.Skip(skip).Take(take);
+
+            while (batchedContentItems.Any())
+            {
+                // Preload all the versions for this batch from the database.
+                var versionIds = batchedContentItems
+                    .Where(x => !String.IsNullOrEmpty(x.ContentItemVersionId))
+                    .Select(x => x.ContentItemVersionId);
+
+                var versions = await _session
+                    .Query<ContentItem, ContentItemIndex>(x => x.ContentItemVersionId.IsIn(versionIds))
+                    .ListAsync();
+
+                foreach (var version in versions)
+                {
+                    await LoadAsync(version);
+                }
+
+                // Query for versions that may be evicted.
+                var publishedLatestVersions = versions.Where(x => x.Latest && x.Published);
+                var reducedContentItems = batchedContentItems.Except(publishedLatestVersions);
+                var evictionIds = reducedContentItems.Select(x => x.ContentItemId);
+
+                var evictionVersions = await _session.Query<ContentItem, ContentItemIndex>()
+                    .Where(x => x.ContentItemId.IsIn(evictionIds) && (x.Latest || x.Published))
+                    .ListAsync();
```

Comment:
The reason for the two queries here, is that the first is against a `ContentItemVersionId`, to get the existing versions, that are going to be updated.

Then the second is against a `ContentItemId` to get the versions that are in the database that are latest and/or published.

So I could not see a clear way to get both of what we need from a single query.

Because a `ContentItemVersion` maybe being updated, but two other versions of the `ContentItemId` may also be published and/or latest, so they are the ones we are looking for in the second query.